packages/stream_video/lib/src/sorting/call_participant_state_sorting.dart

```diff
@@ -53,6 +54,24 @@ int pinned(CallParticipantState a, CallParticipantState b) {
   return 0;
 }
 
+/// A comparator factory which creates a comparator which prioritizes
+/// participants who are from a specific source (e.g., WebRTC, RTMP, WHIP...).
+Comparator<CallParticipantState> byParticipantSource(
+  SfuParticipantSource source,
+) {
+  return (CallParticipantState a, CallParticipantState b) {
+    if (a.participantSource == source && b.participantSource != source) {
+      return -1;
+    }
+
+    if (a.participantSource != source && b.participantSource == source) {
+      return 1;
+    }
+
+    return 0;
+  };
+}
+
```

Comment:
I think this would be a good piece of code to unit test.
I wonder if somebody would ever want to sort by multiple sources, but I guess not.